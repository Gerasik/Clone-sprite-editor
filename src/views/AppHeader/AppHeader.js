export default class AppHeader {
  constructor(className) {
    this.className = className;
    this.block = null;
  }

  render() {
    const header = document.createElement('header');
    header.classList.add(this.className);
    header.innerHTML = `<button class="${this.className}__menu-button"><div></div><div></div><div></div></button>
                        <h1 class="${this.className}__title">Piskel Clone</h1>
                        <button class="${this.className}__position"><div></div><div></div><div></div></button>`;
    document.body.appendChild(header);
    this.block = header;
  }
}
