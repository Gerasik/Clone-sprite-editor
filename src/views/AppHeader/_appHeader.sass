$color: #6f6f6f
$color-text: #545454

.page-header
  width: 100%
  display: flex
  justify-content: space-between
  align-items: center
  border-bottom: 3px solid $color
  box-shadow: 0 0 15px 5px $color
  color: $color-text
  &__menu-button,
  &__position
    margin: 5px 20px
    background-color: inherit
    border: none
    div
      background: $color
      margin: 8px 0px
  &__menu-button
    div
      height: 3px
      width: 30px
  &__title
    font-size: 24px
    flex-grow: 1
  &__position
    div
      height: 5px
      width: 5px
      border-radius: 50%
