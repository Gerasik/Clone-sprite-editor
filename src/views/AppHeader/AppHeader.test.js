import AppHeader from './AppHeader';

describe('new AppHeader', () => {
  it('Should be an instance of Object', () => {
    expect(new AppHeader('header')).toBeInstanceOf(Object);
  });
  it('Should be contain right parameters', () => {
    const newObj = new AppHeader('tools');
    expect(newObj.className).toBe('tools');
    expect(newObj.block).toBe(null);
  });
});

describe('AppHeader.prototype.render', () => {
  it('Should be an instance of Function', () => {
    expect(AppHeader.prototype.render).toBeInstanceOf(Function);
  });

  it('Should be render slide', () => {
    AppHeader.prototype.render();
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
