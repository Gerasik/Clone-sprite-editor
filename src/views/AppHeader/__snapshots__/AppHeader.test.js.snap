// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader.prototype.render Should be render slide 1`] = `
"<header class=\\"undefined\\"><button class=\\"undefined__menu-button\\"><div></div><div></div><div></div></button>
                        <h1 class=\\"undefined__title\\">Piskel Clone</h1>
                        <button class=\\"undefined__position\\"><div></div><div></div><div></div></button></header>"
`;
